src/modules/flow/iio/nodes.c

```diff
@@ -25,6 +25,21 @@
 
 #include <sol-iio.h>
 
+#define ADD_CHANNEL_AXIS(_axis, _nodename) \
+    if (!mdata->iio_base.use_device_default_scale) \
+        channel_config.scale = mdata->scale._axis; \
+    if (!mdata->iio_base.use_device_default_offset) \
+        channel_config.offset = mdata->offset._axis; \
+    mdata->channel_ ## _axis = sol_iio_add_channel(mdata->iio_base.device, _nodename # _axis, &channel_config); \
+    SOL_NULL_CHECK_GOTO(mdata->channel_ ## _axis, error);
+
+#define ADD_CHANNEL_VAL(_val, _nodename) \
+    if (!mdata->iio_base.use_device_default_scale) \
+        channel_config.scale = mdata->scale; \
+    if (!mdata->iio_base.use_device_default_offset) \
+        channel_config.offset = mdata->offset; \
+    mdata->channel_ ## _val = sol_iio_add_channel(mdata->iio_base.device, _nodename, &channel_config);
```

Comment:
Guess this `val` is unnecessary: it's always `val`. You could simply call  `ADD_CHANNEL` and use `mdata->channel_val` directly.